### DISCUSSION TITLE
Does this static site generation copy over public files like favicon, manifest.json etc?

### DISCUSSION THREAD

**Discussion by MeikyuuTrader** [statamic/ssg#82] [Q&A] [answered]
I've used `npm run production and php please ssg:generate` to generate myself a static statamic site, however, it doesn't seem to copy over files from the public folder like favicon or manifest.json. 

Under `storage/app/public` there's only a .gitignore file, is this by design? 

How do you copy a file from public, like manifest.json into the public folder for the static site? I just copying over manifest.json from /public into `storage/app/public` and it doesn't exist. Do I put it into the root `storage/app/static`?

# Edit
Answering my own question, yes, need to copy over favicons, manifest.json and other public files to `storage/app/static`. Not sure if there's an inbuilt command from statamic/ssg for this. 

If you're trying to make a PWA using this [guide](https://gist.github.com/felixwetell/a3c06d1dfae1d3b0fae4968425b75d23) exactly, then the below script (put into package.json) will be helpful to automatically copy over public files and put them into the static location. Run `npm run copy` after `php please ssrg:generate`

`        "copy": "cp public/*.png public/favicon.ico public/manifest.json public/robots.txt public/serviceworker.js storage/app/static/"`

**Comment by duncanmcclean** [comment, marked as answer]
I'd recommend adding the files you wish to copy into the `copy` array inside the SSG config (`config/statamic/ssg.php`):

```
'copy' => [
        public_path('favicon.ico') => '',
        public_path('manifest.json') => '',
       // ...
],
```

**Reply by MeikyuuTrader** [reply]
The above code gives

Error exception
`  copy(): The second argument to copy() function cannot be a directory`

Has to be like 

```
'copy' => [
        public_path('favicon.ico') => 'favicon.ico',
        public_path('manifest.json') => 'manifest.json',
       // ...
],
```

Instead.

**Reply by duncanmcclean** [reply]
I didn't try it so was just giving it a guess. Glad you could sort it!
